pkg/provider/provider.go

```diff
@@ -538,18 +538,35 @@ func (k *kubeProvider) Diff(
 		oldInputs.SetNamespace("")
 		newInputs.SetNamespace("")
 	}
+	if oldInputs.GroupVersionKind().Empty() {
+		oldInputs.SetGroupVersionKind(gvk)
+	}
+
+	patch, _, err := openapi.PatchForResourceUpdate(
+		k.clientSet.DiscoveryClientCached, oldInputs, newInputs, oldLiveState)
+	if err != nil {
+		return nil, err
+	}
+	patchObj := map[string]interface{}{}
+	if err = json.Unmarshal(patch, &patchObj); err != nil {
+		return nil, err
+	}
 
 	// Decide whether to replace the resource.
-	replaces, err := forceNewProperties(oldInputs.Object, newInputs.Object, gvk)
+	replaces, err := forceNewProperties(patchObj, gvk)
 	if err != nil {
 		return nil, err
 	}
 
 	// Pack up PB, ship response back.
 	hasChanges := pulumirpc.DiffResponse_DIFF_NONE
-	diff := gojsondiff.New().CompareObjects(oldInputs.Object, newInputs.Object)
-	if len(diff.Deltas()) > 0 {
+
+	var changes []string
+	if len(patchObj) != 0 {
 		hasChanges = pulumirpc.DiffResponse_DIFF_SOME
+		for k := range patchObj {
+			changes = append(changes, k)
```

Comment:
Linter caught that `changes` isn't used later.

Reply:
Fixed.